### PR TITLE
REF: consolidate casting in groupby agg_series

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/groupby/test_bin_groupby.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ def test_series_grouper():
     grouper = libreduction.SeriesGrouper(obj, np.mean, labels, 2)
     result, counts = grouper.get_result()
 
-    expected = np.array([obj[3:6].mean(), obj[6:].mean()])
+    expected = np.array([obj[3:6].mean(), obj[6:].mean()], dtype=object)
```
Comment (jreback) [inline pandas/tests/groupby/test_bin_groupby.py:23]: these look like internal tests, but is anything user facing changed? (no looks like)? 

why are these now returning object? (e.g. can you just remove the cast_back above)?

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jbrockmendel** [comment]
These tests are for the low-level cython func, and this PR makes the
casting from object take place after the cython call instead of within it.
Nothing user facing.

File: pandas/tests/groupby/test_bin_groupby.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/groupby/test_bin_groupby.py:23]: > (e.g. can you just remove the cast_back above)?

Figuring out how to do that is the next step.  ATM doing that breaks a handful of tests.  This PR doesn't change any user-facing behavior.

Comment (jbrockmendel) [reply pandas/tests/groupby/test_bin_groupby.py:23]: follow-up with this is about ready

**Review by jreback — APPROVED** [review]

**Pull request merged**